src/main/java/io/odpf/depot/http/record/HttpRequestRecord.java

```diff
@@ -0,0 +1,46 @@
+package io.odpf.depot.http.record;
+
+import io.odpf.depot.error.ErrorInfo;
+import io.odpf.depot.http.response.HttpSinkResponse;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+@AllArgsConstructor
+@Getter
+public class HttpRequestRecord {
+
+    protected static final String SUCCESS_CODE_PATTERN = "^2.*";
```

Comment:
we don't need to to parse any response here. This check should move to HttpSinkResponse class.